jsm.go

```diff
@@ -462,6 +465,57 @@ func (js *js) DeleteStream(name string) error {
 	return nil
 }
 
+type apiMsgGetRequest struct {
+	Seq uint64 `json:"seq"`
+}
+
+type storedMsg struct {
+	Subject  string    `json:"subject"`
+	Sequence uint64    `json:"seq"`
+	Header   []byte    `json:"hdrs,omitempty"`
+	Data     []byte    `json:"data,omitempty"`
+	Time     time.Time `json:"time"`
+}
+
+// apiMsgGetResponse is the response for a Stream get request.
+type apiMsgGetResponse struct {
+	apiResponse
+	Message *storedMsg `json:"message,omitempty"`
+	Success bool       `json:"success,omitempty"`
+}
+
+// GetMsg fetches a message from a stream by sequence number.
+func (js *js) GetMsg(name string, seq uint64) (*Msg, error) {
+	if name == _EMPTY_ {
+		return nil, ErrStreamNameRequired
+	}
+
+	req, err := json.Marshal(&apiMsgGetRequest{Seq: seq})
+	if err != nil {
+		return nil, err
+	}
+
+	dsSubj := js.apiSubj(fmt.Sprintf(apiMsgGetT, name))
+	r, err := js.nc.Request(dsSubj, req, js.wait)
+	if err != nil {
+		return nil, err
+	}
+
+	var resp apiMsgGetResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return nil, err
+	}
+	if resp.Error != nil {
+		return nil, errors.New(resp.Error.Description)
+	}
+
+	msg := resp.Message
```

Comment:
Probably should return storedMsg. Need access to timestamp etc. Also below you forget headers.

Reply:
Maybe call it StreamMsg or RawStreamMsg.

Reply:
Updated and renamed it `StreamMsg`.

Reply:
Synced with @ColinSullivan1 and I think now that `RawStreamMsg` might be better avoid confusion with regular `Msg` so updated again. Also it is now decoding the headers from the fetched message and  converting from `[]byte` to `http.Header`.

Reply:
ok